## **Indicadores de desarrollo mundial (WDI) es la principal colección de indicadores de desarrollo del Banco Mundial, compilada a partir de fuentes internacionales oficialmente reconocidas. Presenta los datos de desarrollo global más actualizados y precisos disponibles e incluye estimaciones nacionales, regionales y globales. **
Link: https://databank.worldbank.org/source/world-development-indicators

In [6]:
import pandas as pd
import numpy as np

In [8]:
df = pd.read_csv('/content/e3e76ee2-858c-4e45-b485-40036b3ba2bb_Data.csv')

In [ ]:
df.head()

Country Name Country Code  \
0  Afghanistan          AFG   
1  Afghanistan          AFG   
2  Afghanistan          AFG   
3  Afghanistan          AFG   
4  Afghanistan          AFG   

                                         Series Name           Series Code  \
0            Access to electricity (% of population)        EG.ELC.ACCS.ZS   
1         Adjusted net national income (current US$)        NY.ADJ.NNTY.CD   
2  Adjusted net national income per capita (annua...  NY.ADJ.NNTY.PC.KD.ZG   
3         Adjusted savings: gross savings (% of GNI)     NY.ADJ.ICTR.GN.ZS   
4  Adolescent fertility rate (births per 1,000 wo...           SP.ADO.TFRT   

  1990 [YR1990] 1991 [YR1991] 1992 [YR1992] 1993 [YR1993] 1994 [YR1994]  \
0            ..            ..            ..            ..            ..   
1            ..            ..            ..            ..            ..   
2            ..            ..            ..            ..            ..   
3            ..            ..            ..            ..            ..   
4      162.8354      163.3822       163.929      164.1888      164.4486   

  1995 [YR1995]  ...     2011 [YR2011]     2012 [YR2012]     2013 [YR2013]  \
0            ..  ...  43.2220191955566  69.0999984741211  68.2906494140625   
1            ..  ...  16267323504.6719  18225512193.1932  18502504965.1779   
2            ..  ...                ..                ..                ..   
3            ..  ...                ..                ..                ..   
4      164.7084  ...           107.256           100.797            94.429   

      2014 [YR2014]     2015 [YR2015]     2016 [YR2016]     2017 [YR2017]  \
0              89.5              71.5  97.6999969482422  97.6999969482422   
1  18834093829.3395  17785739175.8072  16937442592.6735  17641172246.3335   
2                ..                ..                ..                ..   
3                ..                ..                ..                ..   
4            88.061            81.693            75.325            68.957   

      2018 [YR2018]     2019 [YR2019]     2020 [YR2020]  
0  96.6161346435547  97.6999969482422  97.6999969482422  
1  16815188084.6708  17645416937.3793  18515111650.4997  
2                ..                ..                ..  
3                ..                ..                ..  
4            65.141            61.325            57.509  

[5 rows x 35 columns]

In [ ]:
df.shape

(27403, 35)

In [ ]:
# Observamos los tipos de datos que manejamos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27403 entries, 0 to 27402
Data columns (total 35 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Country Name   27400 non-null  object
 1   Country Code   27398 non-null  object
 2   Series Name    27398 non-null  object
 3   Series Code    27398 non-null  object
 4   1990 [YR1990]  27398 non-null  object
 5   1991 [YR1991]  27398 non-null  object
 6   1992 [YR1992]  27398 non-null  object
 7   1993 [YR1993]  27398 non-null  object
 8   1994 [YR1994]  27398 non-null  object
 9   1995 [YR1995]  27398 non-null  object
 10  1996 [YR1996]  27398 non-null  object
 11  1997 [YR1997]  27398 non-null  object
 12  1998 [YR1998]  27398 non-null  object
 13  1999 [YR1999]  27398 non-null  object
 14  2000 [YR2000]  27398 non-null  object
 15  2001 [YR2001]  27398 non-null  object
 16  2002 [YR2002]  27398 non-null  object
 17  2003 [YR2003]  27398 non-null  object
 18  2004 [YR2004]  27398 non-n

Primero vamos a observar los datos que nos brinda cada columna

In [ ]:
df['Country Name'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'American Samoa', 'Andorra',
       'Angola', 'Antigua and Barbuda', 'Argentina', 'Armenia', 'Aruba',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas, The', 'Bahrain',
       'Bangladesh', 'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin',
       'Bermuda', 'Bhutan', 'Bolivia', 'Bosnia and Herzegovina',
       'Botswana', 'Brazil', 'British Virgin Islands',
       'Brunei Darussalam', 'Bulgaria', 'Burkina Faso', 'Burundi',
       'Cabo Verde', 'Cambodia', 'Cameroon', 'Canada', 'Cayman Islands',
       'Central African Republic', 'Chad', 'Channel Islands', 'Chile',
       'China', 'Colombia', 'Comoros', 'Congo, Dem. Rep.', 'Congo, Rep.',
       'Costa Rica', "Cote d'Ivoire", 'Croatia', 'Cuba', 'Curacao',
       'Cyprus', 'Czechia', 'Denmark', 'Djibouti', 'Dominica',
       'Dominican Republic', 'Ecuador', 'Egypt, Arab Rep.', 'El Salvador',
       'Equatorial Guinea', 'Eritrea', 'Estonia', 'Eswatini', 'Ethiopia',
       'Faroe Islands',

In [ ]:
# en primera instancia podemos apreciar que el dataset presenta tanto paises como regiones, 
# parece a primera vista que no presenta valores atipicos
df['Country Name'].nunique()

268

In [ ]:
# vemos la cantidad de datos nulos que maneja esta columna
df['Country Name'].isnull().sum()

3

In [ ]:
# Todas los datos/indicadores que presenta
df['Series Name'].unique()

array(['Access to electricity (% of population)',
       'Adjusted net national income (current US$)',
       'Adjusted net national income per capita (annual % growth)',
       'Adjusted savings: gross savings (% of GNI)',
       'Adolescent fertility rate (births per 1,000 women ages 15-19)',
       'Adolescents out of school (% of lower secondary school age)',
       'Children out of school, primary',
       'Cause of death, by communicable diseases and maternal, prenatal and nutrition conditions (% of total)',
       'Cause of death, by non-communicable diseases (% of total)',
       'Condom use, population ages 15-24, male (% of males ages 15-24)',
       'Condom use, population ages 15-24, female (% of females ages 15-24)',
       'Cost of business start-up procedures (% of GNI per capita)',
       'Cost of business start-up procedures, female (% of GNI per capita)',
       'Cost of business start-up procedures, male (% of GNI per capita)',
       'Death rate, crude (per 1,000 pe

In [ ]:
df.describe()

Country Name Country Code                              Series Name  \
count         27400        27398                                    27398   
unique          268          266                                      103   
top     Afghanistan          AFG  Access to electricity (% of population)   
freq            103          103                                      266   

           Series Code 1990 [YR1990] 1991 [YR1991] 1992 [YR1992]  \
count            27398         27398         27398         27398   
unique             103          9101         10138         10361   
top     EG.ELC.ACCS.ZS            ..            ..            ..   
freq               266         17190         16038         15913   

       1993 [YR1993] 1994 [YR1994] 1995 [YR1995]  ... 2011 [YR2011]  \
count          27398         27398         27398  ...         27398   
unique         10645         10858         11733  ...         15222   
top               ..            ..            ..  ...            ..   
freq           15660         15488         14688  ...          9370   

       2012 [YR2012] 2013 [YR2013] 2014 [YR2014] 2015 [YR2015] 2016 [YR2016]  \
count          27398         27398         27398         27398         27398   
unique         15271         15330         15484         16741         15654   
top               ..            ..            ..            ..            ..   
freq            9250          9087          8926          7601          8711   

       2017 [YR2017] 2018 [YR2018] 2019 [YR2019] 2020 [YR2020]  
count          27398         27398         27398         27398  
unique         15641         15369         15647         12535  
top               ..            ..            ..            ..  
freq            8755          9018          8701         14068  

[4 rows x 35 columns]

In [ ]:
((df.isnull().sum())*100 / len(df))

Country Name     0.010948
Country Code     0.018246
Series Name      0.018246
Series Code      0.018246
1990 [YR1990]    0.018246
1991 [YR1991]    0.018246
1992 [YR1992]    0.018246
1993 [YR1993]    0.018246
1994 [YR1994]    0.018246
1995 [YR1995]    0.018246
1996 [YR1996]    0.018246
1997 [YR1997]    0.018246
1998 [YR1998]    0.018246
1999 [YR1999]    0.018246
2000 [YR2000]    0.018246
2001 [YR2001]    0.018246
2002 [YR2002]    0.018246
2003 [YR2003]    0.018246
2004 [YR2004]    0.018246
2005 [YR2005]    0.018246
2006 [YR2006]    0.018246
2007 [YR2007]    0.018246
2008 [YR2008]    0.018246
2009 [YR2009]    0.018246
2010 [YR2010]    0.018246
2011 [YR2011]    0.018246
2012 [YR2012]    0.018246
2013 [YR2013]    0.018246
2014 [YR2014]    0.018246
2015 [YR2015]    0.018246
2016 [YR2016]    0.018246
2017 [YR2017]    0.018246
2018 [YR2018]    0.018246
2019 [YR2019]    0.018246
2020 [YR2020]    0.018246
dtype: float64

In [9]:
nulos_total = df.isnull().sum()
nulos_total

Country Name     3
Country Code     5
Series Name      5
Series Code      5
1990 [YR1990]    5
1991 [YR1991]    5
1992 [YR1992]    5
1993 [YR1993]    5
1994 [YR1994]    5
1995 [YR1995]    5
1996 [YR1996]    5
1997 [YR1997]    5
1998 [YR1998]    5
1999 [YR1999]    5
2000 [YR2000]    5
2001 [YR2001]    5
2002 [YR2002]    5
2003 [YR2003]    5
2004 [YR2004]    5
2005 [YR2005]    5
2006 [YR2006]    5
2007 [YR2007]    5
2008 [YR2008]    5
2009 [YR2009]    5
2010 [YR2010]    5
2011 [YR2011]    5
2012 [YR2012]    5
2013 [YR2013]    5
2014 [YR2014]    5
2015 [YR2015]    5
2016 [YR2016]    5
2017 [YR2017]    5
2018 [YR2018]    5
2019 [YR2019]    5
2020 [YR2020]    5
dtype: int64

In [13]:
nulos_total = (df.isnull().sum()).sum() / 959105  * 100
print(f'Porcentaje total de valores nulos presentes en el dataframe {nulos_total}% (representados en una escala de 0-100%)')

Porcentaje total de valores nulos presentes en el dataframe 0.01803764968381981% (representados en una escala de 0-100%)


In [ ]:
df[df['Country Name'] == 'United States']

Country Name Country Code  \
21218  United States          USA   
21219  United States          USA   
21220  United States          USA   
21221  United States          USA   
21222  United States          USA   
...              ...          ...   
21316  United States          USA   
21317  United States          USA   
21318  United States          USA   
21319  United States          USA   
21320  United States          USA   

                                             Series Name  \
21218            Access to electricity (% of population)   
21219         Adjusted net national income (current US$)   
21220  Adjusted net national income per capita (annua...   
21221         Adjusted savings: gross savings (% of GNI)   
21222  Adolescent fertility rate (births per 1,000 wo...   
...                                                  ...   
21316  Unemployment, total (% of total labor force) (...   
21317  Unemployment, total (% of total labor force) (...   
21318           Urban population (% of total population)   
21319                                   Urban population   
21320                 Urban population growth (annual %)   

                Series Code       1990 [YR1990]      1991 [YR1991]  \
21218        EG.ELC.ACCS.ZS                 100                100   
21219        NY.ADJ.NNTY.CD    4957200835250.46   5125431776191.72   
21220  NY.ADJ.NNTY.PC.KD.ZG  0.0452544249677942  -0.89332934967365   
21221     NY.ADJ.ICTR.GN.ZS         18.94942471        18.99938492   
21222           SP.ADO.TFRT             57.0142            58.2626   
...                     ...                 ...                ...   
21316     SL.UEM.TOTL.NE.ZS    5.59999990463257   6.80000019073486   
21317        SL.UEM.TOTL.ZS                  ..   6.80000019073486   
21318     SP.URB.TOTL.IN.ZS                75.3             75.701   
21319           SP.URB.TOTL           187966119          191509147   
21320           SP.URB.GROW    1.41025628541903   1.86738439894312   

          1992 [YR1992]      1993 [YR1993]     1994 [YR1994]  \
21218               100                100               100   
21219  5429249744321.47   5668523246585.74  6063645331997.61   
21220  2.01993059129006  0.693391456824969  3.33110527512687   
21221       17.91467024        17.35927344       18.14329233   
21222            59.511            57.8868           56.2626   
...                 ...                ...               ...   
21316               7.5   6.90000009536743  6.11999988555908   
21317               7.5   6.90000009536743  6.11999988555908   
21318            76.097             76.488            76.875   
21319         195199459          198806845         202278113   
21320  1.90863299657144   1.83118227604993   1.7309822295661   

          1995 [YR1995]  ...     2011 [YR2011]     2012 [YR2012]  \
21218               100  ...               100               100   
21219  6413137087392.22  ...    13346185446958  14068464248286.6   
21220  2.29382210850626  ...  1.44753427724325  2.84103034296143   
21221       18.90882017  ...       16.00380139       18.12624377   
21222           54.6384  ...           30.2658            27.937   
...                 ...  ...               ...               ...   
21316  5.65000009536743  ...  8.94999980926514  8.06999969482422   
21317  5.65000009536743  ...  8.94999980926514  8.06999969482422   
21318            77.257  ...            80.944            81.119   
21319         205718394  ...         252208133         254614421   
21320  1.68646666337335  ...  0.93950541093758  0.94956550837067   

           2013 [YR2013]     2014 [YR2014]      2015 [YR2015]  \
21218                100               100                100   
21219   14455936986899.8  15130674681973.4   15781103411449.7   
21220  0.577814877815001  2.18230214623172   3.21435123963356   
21221        18.64080346       19.63508587        19.75042482   
21222            26.3216           24.7062            23.0908   
...                  ... 

In [ ]:
j = []
for i in df.iloc[:,4:35].columns:
    j.append(len(df[df[i] == '..']))
print('Numero total de valores faltantes: ' + str(np.sum(j) ))

Numero total de valores faltantes: 366694


In [ ]:
df.shape

(27403, 35)

In [ ]:
print('Numero total de valores: ' + str(27403* 35))

Numero total de valores: 959105


In [ ]:
print('Porcentaje de valores faltantes: ' + str((366694 / 959105) * 100) + '%')

Porcentaje de valores faltantes: 38.23293591421169%


Breve analisis de la situacion: 
+ Presentamos un porcentaje muy bajo de datos nulos
+ Presentamos un porcentaje relativamente alto de valores faltantes (38%), hay que analizar si puede imposibilitar el analisis de cierto pais/region
+ El dataset presenta mucho ruido por la cantidad de filas y de indicadores presentes en la columna 'Series Name' (103 por cada pais/region), se tendra que evaluar el tipo de analisis que se aplique al mismo



ACLARACION: 


            Valores nulos = NaN
            Valores faltantes = '..' 